### PR TITLE
chore: sauce connect updates

### PR DESCRIPTION
**Pull request by patrickhousley**
Updating sauce connect and implementing platform preferences for sauce browsers.
---

### Overview

- Removed the `sauce-connect-launcher` dependency and switched JIL to use the `saucelabs` dependency.
- Updated the sauce labs connect process to hard code the newer version of the connect proxy recently released by SauceLabs.
- Added filtering to the sauce browser script to apply some preferences regarding the platform version to use for each browser. For example, prefer `Windows 11` for chrome, firefox, and edge.
- IE tests were also failing due to missing babel dependency errors and babel configuration issues in the test server. While the babel docs says those plugins are included in the preset, I had to manually install them and add `loose: true` to get IE tests to all pass without errors and warnings.

### Related Issue(s)

N/A

### Testing

Run the test suite.

### CODE REVIEW

**Comment by patrickhousley** [comment]
The safari test failing in GHA is passing when ran manually.

File: tools/jil/util/external-services.js [hunk 1]
```diff
@@ -28,60 +25,25 @@ function startExternalServices (browsers, config, cb) {
...
 
-  let remaining = [needSauce].filter(Boolean).length
-
-  if (needSauce) startSauce(config, checkDone)
-
-  function checkDone () {
-    if (--remaining) return
-    cb()
-  }
+  import('../../saucelabs/utils.mjs')
```
Comment (metal-messiah) [inline tools/jil/util/external-services.js:28]: Asking for a friend, can you mix `require()` and `import()` in the same file like this?

Comment (patrickhousley) [reply tools/jil/util/external-services.js:28]: You have to use `import()` since this is a CJS file and its pulling in an ESM file.

**Review by metal-messiah — APPROVED** [review]

**Review by bjfield — APPROVED** [review]

File: tools/browsers-lists/sauce-browsers.mjs [hunk 2]
```diff
@@ -69,8 +69,6 @@ const minSupportedVersion = apiName => {
     case 'safari':
       return Math.floor(browserslistMinVersion('last 10 Safari versions'))
     case 'android':
-      // browserslist only ever provides the most recent ChromeAndroid version.
-      // Sauce Labs provides only a single Chrome Android version (100) on all emulators.
```
[the remark this answers is not in the document]
Comment (patrickhousley) [reply tools/browsers-lists/sauce-browsers.mjs:73]: I checked different versions of Android and they have different versions of chrome on them. That said, you can't actually pick which version of chrome in android to use, just which version of android. Once we switch over 100% to WDIO we can remove the limit on android and ios versions being tested.

File: tools/saucelabs/utils.mjs [hunk 3]
```diff
@@ -39,17 +39,18 @@ export function getSauceConnectTunnelName () {
  */
 export function buildSauceConnectOptions (cliArgs) {
   const opts = {
+    scVersion: '4.9.0',
```
[the remark this answers is not in the document]
Comment (patrickhousley) [reply tools/saucelabs/utils.mjs:42]: I didn't know `'latest'` was a thing. I can change it.

[the remark this answers is not in the document]
Comment (patrickhousley) [reply tools/saucelabs/utils.mjs:42]: This doesn't work for me. I think you may be confusing a browser capabilities property with properties passed to the saucelabs npm package.

**Pull request merged**